### PR TITLE
Enhancement/modeler refactoring

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: pyaedt/modeler/Model2D.py [hunk 1]
```diff
@@ -58,14 +60,17 @@ def __get__(self, instance, owner):
     def primitives(self):
         """Primitives.
 
+        .. deprecated:: 0.4.15
+            No need to use primitives anymore. you can instantiate primitives methods directly from modeler instead.
```
Comment (MaxJPRey) [inline pyaedt/modeler/Model2D.py:64]:
```suggestion
            No need to use primitives anymore. You can instantiate primitives methods directly from modeler instead.
```

File: doc/source/Resources/User_Guide.rst [hunk 2]
```diff
@@ -45,6 +45,184 @@ The same result can be obtained with the following code:
...
+
+.. image:: ./aedt_variables.png
+  :width: 800
+  :alt: Variable Management
+
+
+Variable Manager allows user to create advanced equations and manage them through PyAEDT.
+
+The Variable value can be set or get using application setter and getter but user can access
+more comprehensive set of function by accessing directly the variable_manager object.
```
Comment (MaxJPRey) [inline doc/source/Resources/User_Guide.rst:70]:
```suggestion
more comprehensive set of functions by accessing directly the variable_manager object.
```

File: doc/source/Resources/User_Guide.rst [hunk 3]
```diff
@@ -45,6 +45,184 @@ The same result can be obtained with the following code:
...
+        print(face.center)
+        for edge in face:
+            print(edge.midpoint)
+            for vertice in edge.vertices:
+                print(edge.position)
+     for vertice in box.vertices:
+        print(edge.position)
+
+
+All objects allows to execute any modeler operations like union or subtraction.
```
Comment (MaxJPRey) [inline doc/source/Resources/User_Guide.rst:132]:
```suggestion
All objects allow to execute any modeler operations like union or subtraction.
```

File: doc/source/Resources/User_Guide.rst [hunk 4]
```diff
@@ -45,6 +45,184 @@ The same result can be obtained with the following code:
...
+
+
+.. image:: ./Mesh_Operations.png
+  :width: 800
+  :alt: Mesh object List
+
+Once an object is created or is present in the design (from a loaded project), user can
+simply get the related object using getters. Getter works either with object id or object name.
+Object returned has all features even if it has not been created in PyAEDT.
+Next example shows how easily user can go deeper into edges and vertices of faces or 3d objects.
```
Comment (MaxJPRey) [inline doc/source/Resources/User_Guide.rst:173]: @maxcapodi78 I think this is a copy/paste from the example above for the faces, edges, vertices...
I think we can delete it.

File: doc/source/Resources/User_Guide.rst [hunk 5]
```diff
@@ -45,6 +45,184 @@ The same result can be obtained with the following code:
...
```
Comment (MaxJPRey) [inline doc/source/Resources/User_Guide.rst:173]: @maxcapodi78 I think this is a copy/paste from the example above for the faces, edges, vertices...
I think we can delete it.

File: doc/source/Resources/User_Guide.rst [hunk 6]
```diff
@@ -45,6 +45,184 @@ The same result can be obtained with the following code:
...
+
+.. image:: ./Setups.png
+  :width: 800
+  :alt: Setup Editing and Creation
+
+
+Optimization
+~~~~~~~~~~~~
+Optimetrics allows user to setup parametric analysis, sensitivity analysis, optimization and DOE.
+PyAEDT is able to read all optietrics setup already present in a design and create new ones.
```
Comment (MaxJPRey) [inline doc/source/Resources/User_Guide.rst:206]:
```suggestion
PyAEDT is able to read all optimetrics setup already present in a design and create new ones.
```

File: pyaedt/modeler/Model3DLayout.py [hunk 7]
```diff
@@ -144,10 +147,19 @@ def model_units(self, units):
...
     @property
     def primitives(self):
-        """Primitives."""
-        if self._primitivesDes != self._app.project_name + self._app.design_name:
-            self._primitives = Primitives3DLayout(self)
-            self._primitivesDes = self._app.project_name + self._app.design_name
+        """Primitives.
+
+        .. deprecated:: 0.4.15
+            No need to use primitives anymore. you can instantiate primitives methods directly from modeler instead.
```
Comment (MaxJPRey) [inline pyaedt/modeler/Model3DLayout.py:153]:
```suggestion
            No need to use primitives anymore. You can instantiate primitives methods directly from modeler instead.
```

File: pyaedt/modeler/Model3D.py [hunk 8]
```diff
@@ -33,14 +36,17 @@ def __get__(self, instance, owner):
     def primitives(self):
         """Primitives.
 
+        .. deprecated:: 0.4.15
+            No need to use primitives anymore. you can instantiate primitives methods directly from modeler instead.
```
Comment (MaxJPRey) [inline pyaedt/modeler/Model3D.py:40]:
```suggestion
            No need to use primitives anymore. You can instantiate primitives methods directly from modeler instead.
```

File: _unittest/test_40_3dlayout_edb.py [hunk 9]
```diff
@@ -25,10 +25,8 @@ def setup_class(self):
                 os.path.join(local_path, "example_models", original_project_name + ".aedb"),
                 os.path.join(self.local_scratch.path, test_project_name + ".aedb"),
             )
-            try:
-                self.aedtapp = Hfss3dLayout(self.test_project)
-            except:
-                self.aedtapp = None
+        self.aedtapp = Hfss3dLayout(self.test_project)
+
```
Comment (MaxJPRey) [inline _unittest/test_40_3dlayout_edb.py:29]:
```suggestion
```

File: doc/source/Resources/User_Guide.rst (hunk 5 above)
Comment (maxcapodi78) [reply doc/source/Resources/User_Guide.rst]: Fixed. Thanks

**Pull request merged**